solidity/test/avs/HyperlaneServiceManager.t.sol

```diff
@@ -0,0 +1,470 @@
+
+import {IAVSDirectory} from "../../contracts/interfaces/avs/IAVSDirectory.sol";
+import {Quorum, StrategyParams} from "../../contracts/interfaces/avs/IECDSAStakeRegistryEventsAndErrors.sol";
+import {TestDelegationManager} from "../../contracts/test/avs/TestDelegationManager.sol";
+import {ECDSAStakeRegistry} from "../../contracts/avs/ECDSAStakeRegistry.sol";
+import {TestPaymentCoordinator} from "../../contracts/test/avs/TestPaymentCoordinator.sol";
+
+import {IStrategy} from "../../contracts/interfaces/avs/IStrategy.sol";
+import {ISignatureUtils} from "../../contracts/interfaces/avs/ISignatureUtils.sol";
+import {Enrollment, EnrollmentStatus} from "../../contracts/libs/EnumerableMapEnrollment.sol";
+import {IRemoteChallenger} from "../../contracts/interfaces/avs/IRemoteChallenger.sol";
+
+import {HyperlaneServiceManager} from "../../contracts/avs/HyperlaneServiceManager.sol";
+import {TestRemoteChallenger} from "../../contracts/test/TestRemoteChallenger.sol";
+
+import {EigenlayerBase} from "./EigenlayerBase.sol";
+
+contract HyperlaneServiceManagerTest is EigenlayerBase {
+    HyperlaneServiceManager internal _hsm;
+    ECDSAStakeRegistry internal _ecdsaStakeRegistry;
+    TestPaymentCoordinator internal _paymentCoordinator;
+
+    // Operator info
+    uint256 operatorPrivateKey = 0xdeadbeef;
+    address operator;
+
+    bytes32 emptySalt;
+    uint256 maxExpiry = type(uint256).max;
+    uint256 challengeDelayBlocks = 50400; // one week of eth L1 blocks
+    address invalidServiceManager = address(0x1234);
+
+    function setUp() public {
+        _deployMockEigenLayerAndAVS();
+
+        _ecdsaStakeRegistry = new ECDSAStakeRegistry(delegationManager);
+        _paymentCoordinator = new TestPaymentCoordinator();
+
+        _hsm = new HyperlaneServiceManager(
+            address(avsDirectory),
+            address(_ecdsaStakeRegistry),
+            address(_paymentCoordinator),
+            address(delegationManager)
+        );
+        _hsm.setSlasher(slasher);
+
+        IStrategy mockStrategy = IStrategy(address(0x1234));
+        Quorum memory quorum = Quorum({strategies: new StrategyParams[](1)});
+        quorum.strategies[0] = StrategyParams({
+            strategy: mockStrategy,
+            multiplier: 10000
+        });
+        _ecdsaStakeRegistry.initialize(address(_hsm), 6667, quorum);
+
+        // register operator to eigenlayer
+        operator = vm.addr(operatorPrivateKey);
+        vm.prank(operator);
+        delegationManager.registerAsOperator(
+            IDelegationManager.OperatorDetails({
+                earningsReceiver: operator,
+                delegationApprover: address(0),
+                stakerOptOutWindowBlocks: 0
+            }),
+            ""
+        );
+        // set operator as registered in Eigenlayer
+        delegationManager.setIsOperator(operator, true);
+    }
+
+    event AVSMetadataURIUpdated(address indexed avs, string metadataURI);
```

Comment:
I believe you can share events across files now
https://github.com/ethereum/solidity/releases/tag/v0.8.22

Reply:
Ahh nice. But, can we rely on >=0.8.22 given our contracts can also work for >=0.8.0?